File: src/Scripting/PythonPY/flyweight.py
from dataclasses import dataclass

@dataclass(frozen=True)
class TextStyle:
    font: str
    size: int
    color: str

class StyleFactory:
    def __init__(self):
        self._styles = {}
    def get(self, font, size, color):
        key = (font, size, color)
        if key not in self._styles:
            self._styles[key] = TextStyle(*key)
        return self._styles[key]
    @property
    def count(self):
        return len(self._styles)

factory = StyleFactory()
red1 = factory.get("Inter", 12, "red")
red2 = factory.get("Inter", 12, "red")
blue = factory.get("Inter", 12, "blue")
glyphs = [("A", 1, red1), ("B", 2, red2), ("C", 3, blue)]
print(f"styles={factory.count};shared={str(glyphs[0][2] is glyphs[1][2]).lower()};text={''.join(g[0] for g in glyphs)}")

File: src/Systems/Rust/flyweight.rs
use std::{collections::HashMap, rc::Rc};
#[derive(Debug)] struct TextStyle { font: String, size: u32, color: String }
struct StyleFactory { styles: HashMap<String, Rc<TextStyle>> }
impl StyleFactory {
    fn new() -> Self { Self { styles: HashMap::new() } }
    fn get(&mut self, font:&str,size:u32,color:&str)->Rc<TextStyle>{
        let key=format!("{font}|{size}|{color}");
        self.styles.entry(key).or_insert_with(||Rc::new(TextStyle{font:font.into(),size,color:color.into()})).clone()
    }
    fn count(&self)->usize{self.styles.len()}
}
fn main(){let mut f=StyleFactory::new();let r1=f.get("Inter",12,"red");let r2=f.get("Inter",12,"red");let _b=f.get("Inter",12,"blue");let _=&(r1.font.as_str(),r1.size,r1.color.as_str());println!("styles={};shared={};text=ABC",f.count(),Rc::ptr_eq(&r1,&r2));}
